client/src/components/App.js

```diff
@@ -10,26 +21,106 @@ import Player from './Player'
 import UserPageWrapper from './styles/UserPageWrapper'
 
 class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.audioRef = React.createRef()
+  }
+
+  componentDidMount() {
+    this.setPlaylist()
+  }
+
+  setPlaylist = () => {
+    const { getPlaylist } = this.props
+
+    getPlaylist()
+  }
+
   onClickLogout = () => this.props.logout()
 
+  onClickPlaySong = () => {
+    if (this.audioRef.current.currentTime === 0) {
+      this.props.updateSongPlays(this.props.currentSong.id)
+    }
+
+    this.audioRef.current.play()
+    this.props.togglePlaying()
+  }
+
+  onClickPauseSong = () => {
+    this.audioRef.current.pause()
+    this.props.togglePlaying()
+  }
+
+  onClickNextSong = () => {
+    this.props.incrementCurrentIndex()
+
+    setTimeout(() => {
```

Comment:
Using setTimeout almost always points to the wrong execution order or not waiting for something to be finished, it's better to use callbacks or promises. Please investigate a bit more and resolve this in a proper manner. Try using componentDidUpdate?